docs/streams/kafka_on_gdn/kafka-java-client.md

```diff
@@ -0,0 +1,125 @@
+props.put("bootstrap.servers", "<my-gdn>.<my-paas>.macrometa.io:9092");
+Note: The topic/stream in GDN could be either global or local that was mentioned above in username variable.
+```
+
+If you use TLS authentication, then you should add or replace properties are as follows.
+
+```java
+props.put("security.protocol", "SASL_SSL");
+props.put("ssl.truststore.location", "<path-to-file-client.truststore.jks>");
+props.put("ssl.truststore.password", "<truststore-password>");
+props.put("ssl.endpoint.identification.algorithm", "");
+props.put("bootstrap.servers", "<my-gdn>.<my-paas>.macrometa.io:9093");
+```
+
+### Producer
+
+In Kafka, producers write messages to topics.
+
+Also, you need to add additional properties for Producer for specifying a subscription and serializers are as follows.
+
+```java
+props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");
+```
+
+Once you've configured Kafka Properties, you can create a Producer and sent a message for a specific Kafka topic. The topic should have prefix of global either local namespace. A full topic name: `<c8globals or c8locals>.<my-topic>`.
+
+```java
+Producer<String, String> producer = new KafkaProducer<String, String>(props);
+producer.send(new ProducerRecord<String, String>("<c8globals or c8locals>.<my-topic>", "<message-key-1>", "<message-value-1>"));
+```
+
+Make sure that you close your producer when you do not need it.
+
+```java
+producer.close();
+```
+
+### Consumer
+
+In Kafka, consumers subscribe to topics and handle messages that producers publish to those topics. You can instantiate a new consumer by first instantiating a Properties object and passing it properties (as above).
+
+Also, you need to add additional properties for Consumer for specifying a subscription and serializers are as follows.
+
+```java
+props.put("enable.auto.commit", "false");
+props.put("group.id", <my-subscription>);
+props.put("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
+props.put("value.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
+```
+
+Once you've configured a Properties object, you can create a Consumer by specifying a topic or multiple topics with global either local distribution.
+
+For example:
+
+```java
+Arrays.asList("c8globals.topic-1", "c8locals.topic-2").
+
+KafkaConsumer<String, String> consumer = new KafkaConsumer<String, String>(props);
+consumer.subscribe(Arrays.asList("<c8globals or c8locals>.<my-topic>"));
+```
+
+The subscribe method will auto subscribe the consumer to the specified topic and subscription. One way to make the consumer listen on the topic is to set up a while loop. In this example loop, the consumer listens for messages, prints the contents of any received message, and then acknowledges that the message has been processed by commitAsync() method.
```

Comment:
```suggestion
The subscribe method automatically subscribes the consumer to the specified topic and subscription.

One way to make the consumer listen to the topic is to set up a while loop. In this example loop, the consumer listens for messages, prints the contents of any received message, and then acknowledges that the message has been processed by the `commitAsync()` method.
```

Reply:
I suggest putting the first sentence above the code block, combining it with the example intro.

For example, this subscription method automatically subscribes the consumer to the specified topic and subscription.